snippet_parser/core_test.py

```diff
@@ -146,5 +146,42 @@ def test_no_wikicode_in_section_titles(self):
             "== Section title ''with Wikicode'' == \n\nIrrelevant content{{cn}}")
         self.assertEqual(section, 'Section title with Wikicode')
 
+    def _do_extract_citationdetective(self, html, wikitext='{{ cn }}',
```

Comment:
Minor point but I'd just name this `_do_extract_sentences`, to reflect the method it is actually calling under the hood.